packages/core/src/State.ts

```diff
@@ -95,15 +99,18 @@ export class State<
    * @param stateValue
    * @param context
    */
-  public static from<TC, TE extends EventObject = EventObject>(
-    stateValue: State<TC, TE, any> | StateValue,
-    context?: TC | undefined
-  ): State<TC, TE, any> {
+  public static from<
+    TContext extends MachineContext,
+    TEvent extends EventObject = EventObject
+  >(
+    stateValue: State<TContext, TEvent, any> | StateValue,
+    context?: TContext | undefined
+  ): State<TContext, TEvent, any> {
     if (stateValue instanceof State) {
       if (stateValue.context !== context) {
-        return new State<TC, TE>({
+        return new State<TContext, TEvent>({
           value: stateValue.value,
-          context: context as TC,
+          context: context as TContext,
```

Comment:
Since `context` can be `undefined` here and this is passed to the `State`'s constructor I think we should default to an empty object within that constructor as well. This would avoid `state.context` from being potentially `undefined` in more cases

Reply:
5cd04ee